### PR TITLE
Display one decimal for metrics smaller than 10

### LINKED ISSUE #3509: Display decimals when over 1 million
:clipboard: **Description**

Once downloads hit 1 million, the resolution of downloads is simply an integer, e.g. 1M. Pretty big milestone for some projects, but getting to display 2M might take just as long if not longer. What about showing a single decimal digit, e.g. 1.2M which still fits within limits. Perhaps this drops off after 10M. Same would go for billions, etc.

### PR DESCRIPTION
**Pull request by PyvesB**
Replaces #3534 and closes #3509. Has made things a bit more complex but hopefully delivers the desired functionality.

### CODE REVIEW

**Comment by PyvesB** [comment]
A few related test failures on some of the GitHub badges which were using a different metric validator, will fix tomorrow.

**Comment by PyvesB** [comment]
The test failures are now unrelated.

File: services/text-formatters.spec.js [hunk 1]
```diff
@@ -42,14 +42,24 @@ describe('Text formatters', function() {
...
+    given(9950000000001).expect('10T')
     given(4000000000000001).expect('4P')
+    given(4200000000000001).expect('4.2P')
+    given(7100700010058000200).expect('7.1E')
     given(71007000100580002000).expect('71E')
     given(1000000000000000000000).expect('1Z')
-    given(2222222222222222222222222).expect('2Y')
+    given(1100000000000000000000).expect('1.1Z')
+    given(2222222222222222222222222).expect('2.2Y')
+    given(22222222222222222222222222).expect('22Y')
```
Comment (calebcartwright) [inline services/text-formatters.spec.js]: Can't wait til one of my packages has this many downloads 😄

Comment (PyvesB) [reply services/text-formatters.spec.js]: Share some links, I'm sure the maintainers will be happy to download them a few times and help towards that goal! 😁

**Comment by PyvesB** [comment]
"Required statuses must pass before merging" seemed blocked on pending even though all required ones were there, hopefully closing and reopening will sort this out.

**Pull request merged**